### PR TITLE
Isolated changes to add VAD

### PR DESCRIPTION
**Pull request by chadbailey59**
I'm dismantling my [mega-PR](https://github.com/daily-co/daily-ai-sdk/pull/29) into the parts we actually want.

One thing I'm concerned about with this PR: we have a `speaker_enabled` property in the base transport that enables receiving audio. VAD also needs to receive audio, and daily-python doesn't seem to like two things in different threads both trying to get audio frames. For now, I'm 'hijacking' the `speaker_enabled` property to turn on VAD, since we only use it in the whisper local transcription example. But maybe we should have a separate property to enable VAD, and you can't use both at the same time?

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -13,10 +13,13 @@ dependencies = [
     "fal",
     "faster_whisper",
     "google-cloud-texttospeech",
+    "numpy",
     "openai",
     "Pillow",
     "pyht",
     "python-dotenv",
+    "torch",
+    "pyaudio",
```
Comment (Moishe) [inline pyproject.toml:22]: pyaudio introduces an additional outside-of-pypi dependency (you've gotta `brew insall portaudio` on macos). I'm a little on-the-fence about requiring this for the "standard" SDK. Do we think everyone will want VAD, so it's okay to take on this dependency?

**Review by Moishe — COMMENTED** [review]
Before I go too far on this review, I have a kind of high-level thought: would this work better as a service, rather than as a built-in part of the base transport?

If we have a way to merge queues (we will with the queue PR I'm working on), we could merge the VAD queue and the transport input queue.

This would let us keep this very specific logic out of the base transport class, and would let us minimize dependencies so people don't need to install the model and portaudio pieces if they don't want/need this implementation of VAD.

File: pyproject.toml (hunk 1 above)
Comment (chadbailey59) [reply pyproject.toml:22]: I think VAD is going to be pretty instrumental to anybody wanting to do conversations, but I hear your concern. Should we do some kind of [conditional import](https://stackoverflow.com/questions/3496592/conditional-import-of-modules-in-python) and add a note in the README about the macOS dependency?

File: src/dailyai/services/base_transport_service.py [hunk 2]
```diff
@@ -64,8 +130,11 @@ async def run(self):
         self._frame_consumer_thread.start()
 
         if self._speaker_enabled:
-            self._receive_audio_thread = threading.Thread(target=self._receive_audio, daemon=True)
-            self._receive_audio_thread.start()
+            # TODO-CB: daily-python doesn't like these two listening at the same time
```
Comment (Moishe) [inline src/dailyai/services/base_transport_service.py:133]: It feels like we should be able to write a service that consumes audio frames to do VAD detection in the same way the client-side transcription works.

Comment (chadbailey59) [reply src/dailyai/services/base_transport_service.py]: ugh FINE DAD i'll do it the """right way"""

File: src/dailyai/services/base_transport_service.py [hunk 3]
```diff
@@ -14,8 +19,57 @@
...
+    sound = sound.squeeze()  # depends on the use case
+    return sound
+
+
+FORMAT = pyaudio.paInt16
+CHANNELS = 1
+SAMPLE_RATE = 16000
+CHUNK = int(SAMPLE_RATE / 10)
+
+audio = pyaudio.PyAudio()
```
Comment (aconchillo) [inline src/dailyai/services/base_transport_service.py:64]: where is this being used?

File: src/dailyai/services/base_transport_service.py [hunk 4]
```diff
@@ -14,8 +19,57 @@
...
+def int2float(sound):
+    abs_max = np.abs(sound).max()
+    sound = sound.astype('float32')
+    if abs_max > 0:
+        sound *= 1/32768
+    sound = sound.squeeze()  # depends on the use case
+    return sound
+
+
+FORMAT = pyaudio.paInt16
```
Comment (aconchillo) [inline src/dailyai/services/base_transport_service.py:59]: where is `FORMAT` being used?

File: pyproject.toml (hunk 1 above)
Comment (aconchillo) [reply pyproject.toml:22]: why is pyaudio required?

**Pull request merged**